### PR TITLE
Added version in flyteadmin

### PR DESCRIPTION
**Pull request by evalsocket**
# TL;DR
Added GetVersion GRPC Call

## Type
 - [ ] Bug Fix
 - [x] Feature
 - [ ] Plugin

## Are all requirements met?

 - [x] Code completed
 - [ ] Smoke tested
 - [ ] Unit tests added
 - [ ] Code documentation added
 - [ ] Any pending items have an associated Issue

## Tracking Issue
https://github.com/lyft/flyte/issues/173

### CODE REVIEW

File: pkg/manager/mocks/version.go [hunk 1]
```diff
@@ -0,0 +1,20 @@
+package mocks
```
Comment (katrogan) [inline pkg/manager/mocks/version.go:1]: nit: mind using mockery e.g. https://github.com/flyteorg/flyteadmin/blob/c80c8d8fd97f607da83db3ebb84cf0aec3b1f191/pkg/runtime/interfaces/quality_of_service_configuration.go#L11

see https://github.com/flyteorg/flyte/issues/149

File: version/version.go [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+import (
+	"time"
+
+	"github.com/sirupsen/logrus"
+)
+
+// This module provides the ability to inject Build (git sha) and Version information at compile time.
+// To set these values invoke go build as follows
+
+// go build -ldflags “-X github.com/flyteorg/flyteadmin/version.Build=xyz -X github.com/flyteorg/flyteadmin/version.Version=1.2.3"
```
Comment (katrogan) [inline version/version.go:12]: very cool! can we add this here https://github.com/flyteorg/flyteadmin/blob/master/Makefile#L28 so it gets used in the release image https://github.com/flyteorg/flyteadmin/blob/master/Dockerfile#L21

**Review by katrogan — APPROVED** [review]

File: version/version.go (hunk 2 above)
Comment (kumare3) [reply version/version.go:12]: @evalsocket @katrogan, why do we need to copy this here? This exists in flytestdlib, can we not just reuse that?

Comment (katrogan) [reply version/version.go:12]: oh I did not know this! Yes let's re-use it

**Comment by evalsocket** [comment]
@kumare3 @katrogan pr is ready you can review. I tested version service call in flytectl

File: pkg/manager/impl/version_manager_test.go [hunk 3]
```diff
@@ -0,0 +1,15 @@
...
+	"context"
+	"testing"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestVersionManager_GetVersion(t *testing.T) {
+	versionManager := NewVersionManager()
+	_, err := versionManager.GetVersion(context.Background(), &admin.GetVersionRequest{})
```
Comment (katrogan) [inline pkg/manager/impl/version_manager_test.go:13]: nit: can we set the build, version, etc in the VersionManager and check that it does get passed in the response obj?

File: pkg/rpc/adminservice/version.go [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+
+import (
+	"context"
+
+	"github.com/flyteorg/flyteidl/gen/pb-go/flyteidl/admin"
+)
+
+func (m *AdminService) GetVersion(ctx context.Context, versionRequest *admin.GetVersionRequest) (*admin.GetVersionResponse, error) {
+
+	response, err := m.VersionManager.GetVersion(ctx, versionRequest)
```
Comment (katrogan) [inline pkg/rpc/adminservice/version.go:11]: can we add a catch for panics like here: https://github.com/flyteorg/flyteadmin/blob/master/pkg/rpc/adminservice/project.go#L17?

**Review by katrogan — APPROVED** [review]

File: .goreleaser.yml [hunk 5]
```diff
@@ -7,6 +7,9 @@ builds:
     env:
       - CGO_ENABLED=0
     main: ./cmd/main.go
+    ldflags:
+      - -s -w -X github.com/flyteorg/flyteadmin/version.Version={{.Version}} -X github.com/flyteorg/flyteadmin/version.Build={{.ShortCommit}} -X github.com/flyteorg/flyteadmin/version.BuildTime={{.Date}}
```
Comment (EngHabu) [inline .goreleaser.yml:11]: shouldn't these say stdlib?

```suggestion
      - -s -w -X github.com/flyteorg/flyteadmin/version.Version={{.Version}} -X github.com/flyteorg/flytestdlib/version.Build={{.ShortCommit}} -X github.com/flyteorg/flytestdlib/version.BuildTime={{.Date}}
```

**Review by katrogan — APPROVED** [review]

**Pull request merged**